tests/Codec/LazyBSONArrayCodecTest.php

```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace MongoDB\Tests\Codec;
+
+use MongoDB\BSON\PackedArray;
+use MongoDB\Codec\LazyBSONArrayCodec;
+use MongoDB\Exception\UnsupportedValueException;
+use MongoDB\Model\LazyBSONArray;
+use MongoDB\Tests\TestCase;
+
+class LazyBSONArrayCodecTest extends TestCase
+{
+    private const ARRAY = [
+        'bar',
+        ['bar' => 'baz'],
+        [0, 1, 2],
+    ];
+
+    public function testDecode(): void
+    {
+        $array = (new LazyBSONArrayCodec())->decode($this->getTestArray());
+
+        $this->assertInstanceOf(LazyBSONArray::class, $array);
+        $this->assertSame('bar', $array[0]);
+    }
+
+    public function testDecodeWithWrongType(): void
+    {
+        $codec = new LazyBSONArrayCodec();
+
+        $this->expectExceptionObject(UnsupportedValueException::invalidDecodableValue('foo'));
```

Comment:
Ah, this seems like a more optimal way to test our exceptions than splitting up the class and messages into separate assertions (which entails duplicating the exception message).